src/main/java/backend/model/sponsoring/SponsoringService.kt

```diff
@@ -1,28 +1,29 @@
 package backend.model.sponsoring
 
+import backend.model.event.Event
 import backend.model.event.Team
 import backend.model.user.Sponsor
 import org.javamoney.moneta.Money
 import org.springframework.security.access.prepost.PreAuthorize
 
 interface SponsoringService {
 
-    fun createSponsoring(sponsor: Sponsor, team: Team, amountPerKm: Money, limit: Money): Sponsoring
+    fun createSponsoring(event: Event, sponsor: Sponsor, teams: MutableSet<Team>, amountPerKm: Money, limit: Money): Sponsoring
     fun findByTeamId(teamId: Long): Iterable<Sponsoring>
     fun findBySponsorId(sponsorId: Long): Iterable<Sponsoring>
     fun findOne(id: Long): Sponsoring?
 
-    @PreAuthorize("#sponsoring.team.isMember(authentication.name)")
+    @PreAuthorize("#sponsoring.isTeamMember(authentication.name)")
     fun acceptSponsoring(sponsoring: Sponsoring): Sponsoring
 
-    @PreAuthorize("#sponsoring.team.isMember(authentication.name)")
+    @PreAuthorize("#sponsoring.isTeamMember(authentication.name)")
     fun rejectSponsoring(sponsoring: Sponsoring): Sponsoring
 
     @PreAuthorize("#sponsoring.checkWithdrawPermissions(authentication.name)")
     fun withdrawSponsoring(sponsoring: Sponsoring): Sponsoring
 
-    @PreAuthorize("#team.isMember(authentication.name)")
-    fun createSponsoringWithOfflineSponsor(team: Team, amountPerKm: Money, limit: Money, unregisteredSponsor: UnregisteredSponsor): Sponsoring
+    @PreAuthorize("#unregisteredSponsor.team.isMember(authentication.name)")
```

Comment:
We need to check how this behaves in the real world. Perhaps we might need to check `isTeamMember` instead.